### PR TITLE
Various of fixes for problems found by CodeQL

### PR DESCRIPTION
**Pull request by TrueBrain**

### CODE REVIEW

File: webclient/helpers.py [hunk 1]
```diff
@@ -97,4 +97,4 @@ def redirect(*args, **kwargs):
 
 
 def not_found():
-    flask.abort(redirect("root", message="Data not found"))
+    return flask.abort(redirect("root", message="Data not found"))
```
Comment (LordAro) [inline webclient/helpers.py:100]: Maybe a comment mentioning that this doesn't actually return anything?

Comment (TrueBrain) [reply webclient/helpers.py:100]: It does return something, funny enough. It just, in execution, never get to returning it, as there is a `__call__` in the abort function, that raises an error. Flask can be a bit weird :P

Comment (TrueBrain) [reply webclient/helpers.py:100]: Now I think about it again, it still annoys me, as what flask does is an anti-pattern ... `raise flask.abort` would have been so much more clean ...

Comment (LordAro) [reply webclient/helpers.py:100]: ¯\\\_(ツ)_/¯ Bit disappointing the analyser doesn't recognise functions that always raise exceptions as "exit points" tbh. Perhaps there's an annotation comment or something that can be added to indicate it?

Comment (TrueBrain) [reply webclient/helpers.py:100]: This part of flask is just really wrong. You already notice it when you are writing it. Basically:

```
flask.abort(..)
print("Aborting")
```

Sounds like code that could execute. But the `print` never does. So the SAST is not to blame here, it really is flask that is pushing even the human off the trail. This is why I am in doubt what the right approach is:

1) act like it does return, and let flask figure this out on its own
2) don't return anything, and add comments to indicate this behavior

I tried both .. 2) feels REALLY icky. 1) is the more sane thing (and what I did in this PR) .. as human now you no longer have to care what flask does .. for you it just feels like it is returning a web-result. Which is in fact what you want with `abort()`. So circling back to this PR: I stand by the result in this PR, I think this is the most clean solution. And I think any comment would only throw off the human reader into this sink-hole again :)

Other suggestions?

Comment (TrueBrain) [reply webclient/helpers.py:100]: (in other words, I consider the fact that `flask.abort()` doesn't really return, an implementation detail of flask)

**Review by LordAro — APPROVED** [review]

**Pull request merged**